File: 376-wiggle-subsequence/376-wiggle-subsequence.cpp
class Solution {
public:
    int wiggleMaxLength(vector<int>& nums) {
        int peak = 1, valley = 1;
        if(nums.size() == 1) return 1;
        
        for(int i = 0 ; i < nums.size()-1 ;i++){
            if(nums[i] > nums[i+1]) peak = valley + 1;
            else if(nums[i] < nums[i+1]) valley = peak + 1;
        }
        
        return max(valley, peak);
    }
};

File: 376-wiggle-subsequence/376-wiggle-subsequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var wiggleMaxLength = function(nums) {
    
    var valley = 1 , peak = 1;
    
    for(var i = 0; i < nums.length-1; i++){
        if(nums[i] < nums[i+1]) valley = peak + 1;
        else if (nums[i] > nums[i+1]) peak = valley + 1; 
    }
    
    return Math.max(valley, peak);
};